src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -583,6 +579,10 @@ public String publishText(String text) {
     return text;
   }
 
+  public String publishRequest(String text) {

```

Comment:
This will cause a null pointer exception if text is null.   You should not be calling "trim" here.   Please remove the trim() statement.  "publish" methods should not alter the data that they are publishing.